Services/Mail/classes/class.ilPDMailGUI.php

```diff
@@ -1,6 +1,8 @@
-<?php
+<?php declare(strict_types=1);
 /* Copyright (c) 1998-2012 ILIAS open source, Extended GPL, see docs/LICENSE */
 
+use Psr\Http\Message\RequestInterface;
```

Comment:
IMO `ServerRequestInterface` is more appropriate here (and for the other classes where you introduced it).